### PR TITLE
refactor: make weight argument in xtokens transfers optional

### LINKED ISSUE #838: Replace weight argument in xtokens::transfer with max_fee
Related to https://github.com/open-web3-stack/open-runtime-module-library/issues/656.

In practice, I believe people just put some arbitrary high amount for the weight in `xtokens::transfer`. To me it does make sense to have this as an argument at all. Users do not care how much weight is used on the destination chain - it is simply a representation of computation time on the remote chain. As such, it makes more sense to me to either have the `weight` argument be optional, or to remove it altogether. If not supplied, the code would set the limit to `Unlimited`.

If anything, users may care about how many fees they pay. So we could consider adding an optional `max_fee` argument to `xtokens::transfer`. Note that the `BuyExecution` instruction takes both a max_fee and a max_weight argument. However, even the use `max_fee` is dubious. If the fee is insufficient, presumably the transferred tokens get trapped and the user can then recover them using a `ClaimAsset` instruction. However, the user would still end up paying the fees to pay for execution of this new message..

Interested in hearing your thoughts about this.

### PR DESCRIPTION
**Pull request by sander2**
Closes #838.

Make the weight argument in xtokens transfers optional since in practice it's very rare to set a specific limit - most of the time people just set some arbitrary high value. With this change, you can pass `None` to set the BuyExecution limit to `Unlimited`, which is cleaner and less fragile.

### CODE REVIEW

**Review by shaunxw — APPROVED** [review]
LGTM

File: xtokens/src/lib.rs [hunk 1]
```diff
@@ -763,16 +763,19 @@ pub mod module {
...
 				.map_err(|_| Error::<T>::CannotReanchor)?;
-			Ok(BuyExecution {
-				fees,
-				weight_limit: WeightLimit::Limited(weight),
-			})
+			let weight_limit = if let Some(limit) = weight {
+				WeightLimit::Limited(limit)
+			} else {
+				WeightLimit::Unlimited
+			};
```
Comment (NunoAlexandre) [inline xtokens/src/lib.rs:776]: `nit` but we could make this a little more succint and elegant imo:

```suggestion
			let weight_limit = limit
			  .map(|x| WeightLimit::Limited(x))
			  .unwrap_or(WeightLimit::Unlimited);
```

**Review by NunoAlexandre — COMMENTED** [review]
This looks good but we could also choose to make `dest_weight` of type `WeightLimit` instead of `Option<Weight>` and then having to convert it to `WeightLimit`; that also makes the API more readable by passing `WeightLimit::Unlimited` instead of `None`.

Very happy to have this in either way 🚀

**Comment by sander2** [comment]
> This looks good but we could also choose to make `dest_weight` of type `WeightLimit` instead of `Option<Weight>` and thus having to convert it to `WeightLimit`; doing so would also make the API more readable by passing `WeightLimit::Unlimited` instead of `None`.

Yea that makes sense, I'll try to update the pr later today

**Review by NunoAlexandre — APPROVED** [review]
💯

**Review by xlc — APPROVED** [review]

**Pull request merged**